src/pudl/transform/ferc1.py

```diff
@@ -853,6 +964,228 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+
+        FERC1 XBRL instant period signifies that it is true as of the reported date,
+        while a duration fact pertains to the specified time period. The ``date`` column
+        for an instant fact corresponds to the ``end_date`` column of a duration fact.
+
+        Note: This should always be applied before :meth:``rename_columns``
+
+        Args:
+            raw_xbrl_instant: table representing XBRL instant facts.
+            raw_xbrl_duration: table representing XBRL duration facts.
+
+        Returns:
+            A unified table combining the XBRL duration and instant facts, if both types
+            of facts were present. If either input dataframe is empty, the other
+            dataframe is returned unchanged, except that several unused columns are
+            dropped. If both input dataframes are empty, an empty dataframe is returned.
+        """
+        drop_cols = ["filing_name", "index"]
+        # Ignore errors in case not all drop_cols are present.
+        instant = raw_xbrl_instant.drop(columns=drop_cols, errors="ignore")
+        duration = raw_xbrl_duration.drop(columns=drop_cols, errors="ignore")
+
+        if instant.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL instant table found, returning the "
+                "duration table."
+            )
+            return duration
+        if duration.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL duration table found, returning "
+                "instant table."
+            )
+            return instant
+
+        instant_axes = [
+            col for col in raw_xbrl_instant.columns if col.endswith("_axis")
+        ]
+        duration_axes = [
+            col for col in raw_xbrl_duration.columns if col.endswith("_axis")
+        ]
+        if set(instant_axes) != set(duration_axes):
+            raise ValueError(
+                f"{self.table_id.value}: Instant and Duration XBRL Axes do not match.\n"
+                f"    instant: {instant_axes}\n"
+                f"    duration: {duration_axes}"
+            )
+
+        # Everything above here is the same as the parent function, but here we need
+        # to reshape the instant table such that end-of-last-year becomes
+        # beginning-of-this-year in a separate column. Probably going to need to factor
+        # out the shared vs. different code into separate methods once we know what this
+        # should look like.
+        instant = self.reshape_starting_ending_balances(instant)
+        return pd.concat(
+            [
+                instant.set_index(["report_year", "entity_id"] + instant_axes),
+                duration.set_index(["report_year", "entity_id"] + duration_axes),
+            ],
+            axis="columns",
+        ).reset_index()
+
+    @cache_df("reshape_instant")
+    def reshape_starting_ending_balances(self, df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
I think this method is pretty specific to this particular table, but having a general `reshape_instant_xbrl()` would probably make sense.  Also very open to suggests for more readable / concise / parameterization-friendly methods of reshaping and index manipulation.

Reply:
Replaced this with `process_instant_xbrl()`. We'll see how standardized the operations can be with a few more tables.